### ISSUE TITLE
Gluon / Autograd is_train documentation

### ISSUE THREAD

**Issue by leezu** [apache/incubator-mxnet#7306]
As far as I understand `AutogradRuntime::Get()->IsTraining()` is used throughout the [NDArray API](https://github.com/apache/incubator-mxnet/blob/dad83816511ec04157d2743ee1f220e74a6b0c27/src/c_api/c_api_ndarray.cc) to determine how to set the `OpContext` argument to the functions.

`with autograd.record()`sets the training attribute of AutogradRuntime to True and resets it when leaving the block.

Is this correct? This should be documented. The current Gluon documentation does not state anything about how `OpContext.is_train` is determined, even though this is very relevant to the BatchNorm and Dropout operators. If you confirm my understanding I can add a short note to the Gluon Tutorial. If a documentation page for autograd is added it would make sense to also mention it there.

**Comment by leezu** [comment]
Fixed via https://github.com/apache/incubator-mxnet/pull/7303#issuecomment-319790394 and #7327.
